idaes/apps/flexibility_analysis/decision_rules/dr_config.py

```diff
@@ -0,0 +1,39 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+This module defines a config for specifying options related to decision rules
+"""
+from pyomo.common.config import ConfigDict
+
+
+class DRConfig(ConfigDict):
+    r"""
+    A base class for specifying options for building
+    decision rules.
+    """
+
+    def __init__(
+        self,
+        description=None,
+        doc=None,
+        implicit=False,
+        implicit_domain=None,
+        visibility=0,
+    ):
+        super().__init__(
+            description=description,
```

Comment:
Is the purpose of this just to change the default values for the `implicit` and `implicit_domain` arguments (whilst still leaving them as arguments)?

Reply:
I don't think I changed any defaults. This class is just meant to be a base class for other, related config classes. I guess it is not really necessary. I think I just created it for type hinting.